### PR TITLE
feat: add nuxt template

### PR DESCRIPTION
**Pull request by userquin**
@danielroe maybe we can revert to default nuxt template, this PR using pages, if so I need to extract the logic to a PWABadge sfc.

Nuxt template is under Vue on prompts.

### CODE REVIEW

File: templates/template-nuxt/service-worker/claims-sw.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+// self.__WB_MANIFEST is the default injection point
+precacheAndRoute(self.__WB_MANIFEST)
+
+// clean old assets
+cleanupOutdatedCaches()
+
+let allowlist: RegExp[] | undefined
+// in dev mode, we disable precaching to avoid caching issues
+if (import.meta.env.DEV)
```
Comment (danielroe) [inline templates/template-nuxt/service-worker/claims-sw.ts:17]:
```suggestion
if (import.meta.dev)
```

Comment (userquin) [reply templates/template-nuxt/service-worker/claims-sw.ts]: ahhhhh

File: src/customize/nuxt.ts [hunk 2]
```diff
@@ -1,5 +1,278 @@
...
+          </button>
+          <button @click="$pwa.cancelPrompt()">
+            Close
+          </button>
+        </div>
+      </div>`
+  }
+
+  if (installPWA) {
+    const vif = prompt
```
Comment (userquin) [inline src/customize/nuxt.ts:221]: this should be offline instead prompt

File: src/customize/nuxt.ts [hunk 3]
```diff
@@ -1,5 +1,278 @@
...
+export default defineNuxtConfig({
+  devtools: { enabled: true },
+  nitro: {
+    prerender: {
+      routes: ['/'],
+    },
+  },
+})
+`)
+  addNuxtModule(nuxtConf, '@vite-pwa/nuxt', 'pwa', pwaOptions)
```
Comment (danielroe) [inline src/customize/nuxt.ts:148]: We also need to add `@vite-pwa/nuxt` to the dependencies

Comment (userquin) [reply src/customize/nuxt.ts:148]: also included: https://github.com/vite-pwa/create-pwa/pull/10/files#diff-01a93258c772c4fe47ad08aeec5379ce3a449380f9bdc008c57331618c7b899bR57-R68

Comment (danielroe) [reply src/customize/nuxt.ts:148]: it didn't add it when I was testing... I only had the `resolutions`.

Comment (danielroe) [reply src/customize/nuxt.ts:148]: They are only added when `customServiceWorker` is true. But they should always be added.

File: src/prompts.ts [hunk 4]
```diff
@@ -53,6 +53,7 @@ export const FRAMEWORKS = (<Framework[]>[
         display: 'Nuxt ↗',
         color: lightGreen,
         customCommand: 'npm exec nuxi init TARGET_DIR',
```
Comment (danielroe) [inline src/prompts.ts:55]: You can also pass a custom template to nuxi, which might considerably simplify this... `npx nuxi init TARGET_DIR -t gh:vite-pwa/create-pwa/templates/template-custom-nuxt`

File: src/customize/nuxt.ts (hunk 3 above)
Comment (danielroe) [reply src/customize/nuxt.ts:148]: `dependencies`, I think. It's purely personal preference but I prefer linting, etc. to live in dev deps

File: src/prompts.ts (hunk 4 above)
Comment (userquin) [reply src/prompts.ts:55]: nice 😍

File: src/customize/nuxt.ts (hunk 3 above)
Comment (userquin) [reply src/customize/nuxt.ts:175]: moved to dependencies, only `typescript` and `vue-tsc` in devDependencies

File: src/prompts.ts (hunk 4 above)
Comment (userquin) [reply src/prompts.ts:55]: looks like the command doesn't merge, using `npx nuxi init TARGET_DIR -t gh:vite-pwa/create-pwa/templates/template-custom-nuxt#userquin/feat-add-nuxt-template`:

imagen

Comment (userquin) [reply src/prompts.ts:55]: I'm going to keep it as it is since both templates not being merged

**Pull request merged**